boa/src/builtins/array/mod.rs

```diff
@@ -1043,6 +1043,8 @@ impl Array {
     #[inline]
     pub(crate) fn init(global: &Value) {
         let _timer = BoaProfiler::global().start_event("array", "init");
-        global.set_field("Array", Self::create(global));
+
+        let array = Self::create(global);
+        global.as_object_mut().unwrap().insert_field("Array", array);
```

Comment:
I would use `expect()` instead of `unwrap()` here.

Some extra optimisation that could be done here would be for the `init` function to return a list of `<&str, Value>` pairs, and then add that to the global object in the calling function. This would allow doing this with multiple threads in the future, so that the initialization gets done in paralel. This should probably go in a new PR, though.

Reply:
Made init functions to return a `(&str, Value)` and then just insert in global object. [https://github.com/boa-dev/boa/pull/419/commits/85791b85bc54104c1e7e56b4f18342e034aa1e82 ](https://github.com/boa-dev/boa/pull/419/commits/85791b85bc54104c1e7e56b4f18342e034aa1e82)